App/src/pages/RegularGroups.tsx

```diff
@@ -0,0 +1,83 @@
+import { Container, Table, Thead, Tbody, Tr, Th, Td } from "@chakra-ui/react";
+import { useContext, useEffect, useState } from "react";
+import { GetAllGroups, GroupDTO, MultiGroupResponseDTO } from "../services/group";
+import { UserContext } from "../App";
+import { UserDTO } from "../services/user";
+import Header from "../shared/Header";
+
+export default function DisplayGroups() {
+    const [myGroups, setMyGroups] = useState<GroupDTO[]>([]);
+    const user = useContext(UserContext);
+    const userDTO: UserDTO = { firstName: user.firstName, id: "7", lastName: user.lastName, email: user.email };
+    useEffect(() => {
+        AllGroups();
+        console.log(user);
+    }, []);
+
+    function AllGroups() {
+        GetAllGroups()
+            .then(
+                (res: MultiGroupResponseDTO) => {
+                    setMyGroups(res.data);
+                    setMyGroups(res.data.filter(membership));
```

Comment:
I think this is a mistake here. You set the group data, but then you override it to only show groups that the current user is a part of. Which one is meant to be displayed? All groups or only the current users' groups?

Reply:
Current user's groups, that's why it's being overridden since a method to only grab the groups the user is a member in from the backend doesn't exist at the moment